### PR TITLE
MIDI-23: Prepare a 1D UNet model compatible with ECG dataset

### PR DESCRIPTION
**Pull request by JasiekKaczmarczyk**
[Prepare a 1D UNet model compatible with ECG dataset](https://sacrebleu.atlassian.net/browse/MIDI-23)
Implementation of 1D Unet architecture

I created simple snippet to test if the pipeline is working:
```python
import torch
import matplotlib.pyplot as plt
import seaborn as sns

from models.unet import Unet


if __name__ == "__main__":
    # initializing model
    model = Unet(in_channels=2, out_channels=1, dim=32, dim_mults=(1, 2, 4), kernel_size=9, resnet_block_groups=4)

    # initialing random input
    signal = torch.randn((1, 2, 1000))

    mask_logits = model(signal)
    mask = torch.sigmoid(mask_logits) > 0.5

    x = signal[0, 0, :].detach()
    m = mask[0, 0, :].detach()

    # plot
    fig, axes = plt.subplots(3, 1, figsize=(15, 7))

    sns.lineplot(x, ax=axes[0])
    axes[0].set_title("Signal")

    sns.lineplot(m, ax=axes[1])
    axes[1].set_title("Mask (model not trained)")

    sns.lineplot(x, ax=axes[2])
    sns.lineplot(x * m, alpha=0.8, ax=axes[2])
    axes[2].set_title("Signal + mask")


    plt.tight_layout()
    plt.show()
```

Here is plot of output for untrained model:
model_test_2

### CODE REVIEW

**Comment by roszcz** [comment]
My initial idea was to reimplement the Unet as described in the original Unet paper: https://arxiv.org/abs/1505.04597, and I see that you adjusted the one used in annotated diffusion project. This is great, and we can start training with this, but at the same time original Unet is significantly less complex - there's no attention or resnet blocks involved, so there would be less components to be confused about when something doesn't work.

Here's a plan:
1. Start training experiments with this model,
2. If we're not able to get it to learn quickly (a day or two), let's shift focus to the basic Unet architecture
3. If we can get good results quickly, that's great. It still might be a good idea to still take a look at how performance compares to the "vanilla unet".

For the code itself, I'll take a more in-depth look later today, but from a quick review everything looks great 👍

File: models/modules/attention_layers.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+class MultiHeadAttention(nn.Module):
+    def __init__(self, channels: int, heads: int = 4):
+        """
+        Multi Head Attention Layer
+
+        Args:
+            embedding_size (int): size of embedding dim
+            heads (int): number of attention heads
+        """
```
Comment (roszcz) [inline models/modules/attention_layers.py]: I'm not sure what's the relation between channels and embeddings? 

As far as I can tell, the number of channels is used for this argument, so in the middle where the shape is `[B, 128, 250]`, `embedding_size` is set to 128, which could suggest that we have 250 embeddings each with dim=128, and I'm not sure if this description makes sense. Do you know if there's a good reason to use the name `embedding_size` here?

**Review by roszcz — APPROVED** [review]
Looks great, can't wait to see it train! :train:

File: models/modules/attention_layers.py (hunk 1 above)
Comment (JasiekKaczmarczyk) [reply models/modules/attention_layers.py]: Oh, I forgot to change that. I'll update it, it should be called channels
